main/src/ca/uwaterloo/flix/language/phase/unification/Substitution.scala

```diff
@@ -64,6 +64,10 @@ case class Substitution(m: Map[Type.Var, Type]) {
             case Some(y) if x.kind != t.kind => throw InternalCompilerException(s"Expected kind `${x.kind}' but got `${t.kind}'.")
           }
         case Type.Cst(tc) => t
+        case Type.Apply(Type.Apply(Type.Cst(TypeConstructor.RecordExtend(label)), tpe), rest) =>
+          Type.mkRecordExtend(label, visit(tpe), visit(rest))
+        case Type.Apply(Type.Apply(Type.Cst(TypeConstructor.SchemaExtend(sym)), tpe), rest) =>
+          Type.mkSchemaExtend(sym, visit(tpe), visit(rest))
```

Comment:
I think these two lines are automatically handled by the next case now? Right? So they can be removed.